lib/tictactoe.rb

```diff
@@ -1,58 +1,82 @@
 # frozen_string_literal: true
 
 class TicTacToe
-  def initialize(io:)
-    @board = Array.new(9, ' ')
+  def initialize(io:, board:)
+    @board = board
     @current_token = 'X'
     @io = io
   end
 
   def render_board
-    board_output = <<~BOARD_RENDER 
-    #{@board[0]}|#{@board[1]}|#{@board[2]}
+    board_output = <<~BOARD_RENDER
+    #{@board.get_info_at_position(position: 0)}|#{@board.get_info_at_position(position: 1)}|#{@board.get_info_at_position(position: 2)}
     ------
-    #{@board[3]}|#{@board[4]}|#{@board[5]}
+    #{@board.get_info_at_position(position: 3)}|#{@board.get_info_at_position(position: 4)}|#{@board.get_info_at_position(position: 5)}
     ------
-    #{@board[6]}|#{@board[7]}|#{@board[8]}
+    #{@board.get_info_at_position(position: 6)}|#{@board.get_info_at_position(position: 7)}|#{@board.get_info_at_position(position: 8)}
     BOARD_RENDER
-    io.puts board_output
-    board
+    io.output_to_user board_output
+    board_output
   end
 
-  def place_token(position)
-    board[position] = current_token
-    switch_turn
-    render_board
-  end
+  def cycle_player_turn
+    io.output_to_user "----------------- Player #{current_token} Turn -------------"
+    io.output_to_user 'Select a position to place your token [1 - 9]: '
 
-  def begin_player_turn
     select_another_position_prompt = 'Select another position to place your token [1 - 9]: '
-    desired_position = io.gets.to_i
-    io.puts 'Select a position to place your token [1 - 9]: '
-
-    list_of_accepted_inputs = [*1..9]
-    if !list_of_accepted_inputs.include? desired_position
-      io.puts 'Invalid position'
-      io.puts select_another_position_prompt
-      render_board
-    elsif board[desired_position] == ' '
-      place_token(desired_position)
+    desired_position = io.prompt_user_for_position
+
+    input_is_not_valid = desired_position.nil?
+    if input_is_not_valid
+      io.output_to_user 'Invalid position'
+      io.output_to_user select_another_position_prompt
+    elsif !position_is_free(desired_position)
+      io.output_to_user 'That position is taken!'
+      io.output_to_user select_another_position_prompt
     else
-      io.puts 'That position is taken!'
-      io.puts select_another_position_prompt
-      render_board
+      place_token(desired_position)
+      end_player_turn
     end
+
+    render_board
   end
 
   private
 
   attr_accessor :board, :current_token, :io
 
+  def position_is_free(position)
+    board.get_info_at_position(position: position) == ' '
```

Comment:
What does `' '` mean?

Reply:
I would anticipate this logic being forced to change in an interesting way if, from a display perspective, you were implementing the story to number the available spaces on the board. 

The spread of strings that represent something (`' '`, `'X'`,`'O'`) throughout your code may cause you trouble later for a couple reasons:
- Right now the way you represent the data in your program (representing player tokens and empty spaces) is the same as what you display to the user. If something like the scenario above causes those things to be incompatible, having separate representations will likely be necessary.
- All the possible options for what could be in a board space would be easier to tell if they were defined together in one place. 
- Having references to the strings themselves all over could, in a larger project, make changing that string much harder than if it was defined in one place and used in symbolic form (e.g. as `:x_token` or `PlayerToken::First` or `tokens[:player_one]` etc)